pkg/registry/core/service/strategy.go

```diff
@@ -205,6 +207,50 @@ func topologyKeysInUse(svc *api.Service) bool {
 	return len(svc.Spec.TopologyKeys) > 0
 }
 
+// This is an unusual call.  Service has a number of inter-linked fields and in
+// order to avoid breaking clients we try really hard to infer what users
+// meant.
+//
+// Services are effectively a discriminated union, where type is the
+// discriminator. Some fields just don't make sense with some types.
+func normalizeLinkedFields(newSvc *api.Service, oldSvc *api.Service) {
```

Comment:
Do we want the name to be more specific? That sounds too generic and I'm concerned people would want to insert all kind of changes here. Maybe `wipeNodePortsOnNewType` or something that is very targeted to what we're doing?

Reply:
Well, this is somewhat a trial baloon.  Dual-stack will have to add other fields here, too.  I could be convinced that `wipeIrrelevantFields` and `handleLinkedFields` are 2 different functions if you think that is important.  The main point is that something special is happening